tm2/pkg/bft/privval/signer/remote/client/options.go

```diff
@@ -0,0 +1,143 @@
+	defaultKeepAlivePeriod   = 2 * time.Second
+	defaultRequestTimeout    = 5 * time.Second
+)
+
+// Option is a functional option type used for optional configuration.
+type Option func(*RemoteSignerClient)
+
+// WithDialMaxRetries sets the maximum number of retries when dialing the server.
+// If set to -1 (default), the client will retry indefinitely.
+func WithDialMaxRetries(maxRetries int) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.dialMaxRetries = maxRetries
+	}
+}
+
+// WithDialRetryInterval sets the interval between dial retries when connecting to the server.
+// The default is 5 seconds.
+func WithDialRetryInterval(interval time.Duration) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.dialRetryInterval = interval
+	}
+}
+
+// WithDialTimeout sets the timeout for dialing the server.
+// If set to 0, no timeout is set. The default is 5 seconds.
+func WithDialTimeout(timeout time.Duration) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.dialTimeout = timeout
+	}
+}
+
+// WithKeepAlivePeriod sets the keep alive period for the TCP connection to the server.
+// If set to 0, keep alive is disabled. The default is 2 seconds.
+func WithKeepAlivePeriod(period time.Duration) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.keepAlivePeriod = period
+	}
+}
+
+// WithRequestTimeout sets the timeout for sending requests to the server.
+// If set to 0, no timeout is set. The default is 5 seconds.
+func WithRequestTimeout(timeout time.Duration) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.requestTimeout = timeout
+	}
+}
+
+// WithClientPrivKey sets the private key used by the client to authenticate with the server.
+// The default is a random key.
+func WithClientPrivKey(privKey ed25519.PrivKeyEd25519) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.clientPrivKey = privKey
+	}
+}
+
+// WithAuthorizedKeys sets the list of authorized public keys that the client will accept.
+// If empty (default), all keys are authorized.
+func WithAuthorizedKeys(keys []ed25519.PubKeyEd25519) Option {
+	return func(rsc *RemoteSignerClient) {
+		rsc.authorizedKeys = keys
+	}
+}
+
+// NewRemoteSignerClient creates a new RemoteSignerClient with the required server address and
+// logger. The client can be further configured using functional options.
+func NewRemoteSignerClient(
+	serverAddress string,
+	logger *slog.Logger,
+	options ...Option,
+) (*RemoteSignerClient, error) {
+	// Instantiate a RemoteSignerClient with default options.
+	rsc := &RemoteSignerClient{
+		logger:            logger,
+		dialMaxRetries:    defaultDialMaxRetries,
+		dialRetryInterval: defaultDialRetryInterval,
+		dialTimeout:       defaultDialTimeout,
+		keepAlivePeriod:   defaultKeepAlivePeriod,
+		requestTimeout:    defaultRequestTimeout,
+		clientPrivKey:     ed25519.GenPrivKey(),
+	}
+
+	// Parse the server address.
+	rsc.protocol, rsc.address = osm.ProtocolAndAddress(serverAddress)
+	if rsc.protocol != "tcp" && rsc.protocol != "unix" {
```

Comment:
Nit: these can be constants

Reply:
Done: https://github.com/gnolang/gno/pull/3554/commits/b20f3f7797e7359d0340530cce7042a9919872bc